openupgrade_scripts/scripts/mail/16.0.1.10/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,222 @@
+# NOTHING TO DO
+
+
+mail         / res.partner              / channel_ids (many2many)       : table is now 'mail_channel_member' ('mail_channel_partner')
+# DONE in pre-migration script when renaming the model
+
+---XML records in module 'mail'---
+NEW ir.actions.act_window: mail.act_server_history [renamed from fetchmail module]
+NEW ir.actions.act_window: mail.action_email_server_tree [renamed from fetchmail module]
+NEW ir.actions.act_window: mail.mail_channel_member_action
+NEW ir.actions.act_window: mail.mail_gateway_allowed_action
+NEW ir.actions.act_window: mail.mail_link_preview_action
+NEW ir.actions.act_window: mail.mail_message_schedule_action
+NEW ir.actions.act_window: mail.mail_template_reset_action
+DEL ir.actions.act_window: mail.mail_channel_partner_action
+DEL ir.actions.act_window: mail.mail_resend_cancel_action
+# NOTHING TO DO: noupdate="0" records
+
+NEW ir.cron: mail.ir_cron_mail_gateway_action [renamed from fetchmail module] (noupdate)
+# NOTHING TO DO: modules merged in apriori.py
+
+NEW ir.cron: mail.ir_cron_send_scheduled_message (noupdate)
+# NOTHING TO DO: New noupdate="1" records
+
+NEW ir.model.access: mail.access_fetchmail_server [renamed from fetchmail module]
+NEW ir.model.access: mail.access_mail_channel_member_portal
+NEW ir.model.access: mail.access_mail_channel_member_public
+NEW ir.model.access: mail.access_mail_channel_member_user
+NEW ir.model.access: mail.access_mail_gateway_allowed_system
+NEW ir.model.access: mail.access_mail_link_preview_admin
+NEW ir.model.access: mail.access_mail_message_scheduled_all
+NEW ir.model.access: mail.access_mail_message_scheduled_system
+NEW ir.model.access: mail.access_mail_template_reset
+DEL ir.model.access: mail.access_mail_channel_partner_portal
+DEL ir.model.access: mail.access_mail_channel_partner_public
+DEL ir.model.access: mail.access_mail_channel_partner_user
+DEL ir.model.access: mail.access_mail_resend_cancel
+NEW ir.model.constraint: mail.constraint_mail_activity_check_res_id_is_set
+NEW ir.model.constraint: mail.constraint_mail_channel_channel_type_not_null
+NEW ir.model.constraint: mail.constraint_mail_channel_group_public_id_check
+NEW ir.model.constraint: mail.constraint_mail_channel_member_partner_or_guest_exists
+NEW ir.model.constraint: mail.constraint_mail_channel_rtc_session_channel_member_unique
+NEW ir.model.constraint: mail.constraint_res_users_notification_type
+DEL ir.model.constraint: mail.constraint_mail_channel_partner_partner_or_guest_exists
+DEL ir.model.constraint: mail.constraint_mail_channel_rtc_session_channel_partner_unique
+# NOTHING TO DO: noupdate="0" records
+
+NEW ir.rule: mail.ir_rule_mail_channel_member_group_system (noupdate)
+NEW ir.rule: mail.ir_rule_mail_channel_member_group_user (noupdate)
+# NOTHING TO DO: New noupdate="1" records
+
+DEL ir.rule: mail.ir_rule_mail_channel_partner_group_system (noupdate)
+DEL ir.rule: mail.ir_rule_mail_channel_partner_group_user (noupdate)
+# DONE (post-migration): try to delete
```

Comment:
don't delete. you have to add in the rename_xmlids.

Reply:
These two ir.rules changed (name and domain_force for the second one). They are in noupdate="1" but not present in the noupdate_changes.xml file. Hence just renaming the xmlid is not sufficient to change them. What should I do? Adding them manually to noupdate_changes.xml?

Reply:
Yes, in the `noupdate_changes.xml` you should add:

```xml
  <record id="ir_rule_mail_channel_member_group_system" model="ir.rule">
    <field name="model_id" ref="model_mail_channel_member"/>
    <field name="name">mail.channel.member: admin can manipulate all entries</field>
  </record>
  <record id="ir_rule_mail_channel_member_group_user" model="ir.rule">
    <field name="domain_force">[
                '|',
                    '&amp;',
                        ('channel_id.channel_type', '!=', 'channel'),
                        ('channel_id.is_member', '=', True),
                    '&amp;',
                        ('channel_id.channel_type', '=', 'channel'),
                        '|',
                            ('channel_id.group_public_id', '=', False),
                            ('channel_id.group_public_id', 'in', [g.id for g in user.groups_id])]
            </field>
    <field name="model_id" ref="model_mail_channel_member"/>
    <field name="name">mail.channel.member: write its own entries</field>
  </record>
```